### PR TITLE
Adds Aeon Queue to manage research access requests

### PR DESCRIPTION
**Pull request by winescout**
#### Local Checklist
- [x] Tests written and passing locally?
- [ ] Code style checked?
- [x] QA-ed locally?
- [x] Rebased with `master` branch?
- [x] Configuration updated (if needed)?
- [ ] Documentation updated (if needed)?

#### What does this PR do?
Adds Aeon Queue management for curators
Adds Email to researcher when request appears in queue
Adds rake tasks to approve requests in Aeon Queues
Adds permissions for researchers to view works
Adds revocation of access after 30 days (rake task )
Adds ability to re-authorize access to a researcher

#### Where should a reviewer start?
Check the link identified in the screenshot.

#### Manual testing steps?
- [ ] Create a request in Aeon system for a work
- [ ] Open new Incognito window, and attempt to navigate to a work
- [ ] Login as a curator
- [ ] Navigate to Aeon Queue page
- [ ] Manually move request to approved
- [ ] Check email for link to approved works
- [ ] Open email link in Incognito browser
- [ ] Navigate to work and enjoy access
- [ ] Expire work in regular browser window
- [ ] Attempt to reload page in incognito window, and notice redirect
- [ ] Move Aeon Request back to New -> Approved
- [ ] Navigate in Incognito window to approved works, and notice access
- [ ] repeat process as many times as desired.

#### Screenshots
https://screencast.com/t/yCEp27MfMo8Q
---

#### Database changes

There are a few migrations to add the WorkAuthorization model

@ucsdlib/developers - please review

### CODE REVIEW

**Comment by mcritchlow** [comment]
@lsitu - yes, per our slack team conversation, we'll be deploying the feature/vrr branch to QA for folks here to do testing. Then that branch can be PRd into master once everything is ready. This will make sure we don't end up in a situation where we can't get a bug fix out, or something, while this testing is underway.

File: app/models/concerns/aeon/api_accessor.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'active_support/concern'
+
+module Aeon
+  module ApiAccessor
+    extend ActiveSupport::Concern
+
+    def client
+      # TODO move to ENV vars
+      api_url = 'https://spcoll-request.ucsd.edu/nonshib/api'
+      api_key = 'c4aa6d6a-7d67-4e56-85c5-a704f0749907'
```
[the remark this answers is not in the document]
Comment (winescout) [reply app/models/concerns/aeon/api_accessor.rb]: yep,  its out there :(

**Comment by mcritchlow** [comment]
@winescout @orangewolf - I'm seeing a single Hound violation remaining, though for whatever reason not seeing it here in the UI.. Otherwise seems like we're mostly good to go for getting this merged and deployed in QA.

That said it doesn't look we can do a 'Rebase and merge' here, which is the merge option we usually do. Perhaps things need a rebase on your end against the base branch? It's a lot of commits, so hopefully we can get them in cleanly and you don't have to do (more) work.

**Comment by mcritchlow** [comment]
@winescout @orangewolf - back and looking at this. i'm still seeing that the branch can't be rebased into the current branch due to conflicts w/ the head branch. Given the number of commits, I'm not sure exactly the best strategy for dealing with this. 

Have you been updating your master branch there against our master? If so, perhaps one thing I could do is pull in the latest commits from master into this branch and see if that helps. Thoughts?

**Pull request closed without merging**